docs/deprecation_policy.md

```diff
@@ -0,0 +1,167 @@
+
+A given release of Kubernetes can support any number of API groups and any
+number of versions of each.
+
+The following rules govern the deprecation of elements of the API.  This
+includes:
+   * REST resources (aka API objects)
+   * Fields of REST resources
+   * Enumerated or constant values
+These rules are enforced between official releases, not between
+arbitrary commits to master or release branches.
+
+Rule #1: API elements may only be removed by incrementing the version of the
+API group.
+
+Once an API element has been added to an API group at a particular version, it
+can not be removed from that version, regardless of track.
+
+Note: For historical reasons, we have 2 “monolithic” API groups - “core” (no
+group name) and “extensions”.  We will incrementally move resources from these
+legacy API groups into more domain-specific API groups.
+
+Rule #2: API objects must be able to round-trip between API versions in a given
+release without information loss, with the exception of whole REST resources
+that do not exist in some versions.
+
+For example, an object can be written as v1 and then read back as v2 and
+converted to v1, and the resulting v1 resource will be identical to the
+original.  The representation in v2 might be different from v1, but the system
+knows how to convert between them in both directions.
+
+Rule #3: Other than the “current” API version, older API versions must be
+supported after their announced deprecation for a duration of no less than:
+   * GA: 1 year or 2 releases (whichever is longer)
+   * Beta: 3 months or 1 release (whichever is longer)
+   * Alpha: 0 releases
+
+This is best illustrated by example.  Imagine a Kubernetes release, version X,
+which supports a particular API group.  A new Kubernetes release is made every
+3 months (4 per year).  The following table describes which API versions are
+supported in a series of subsequent releases.
+
+| Release | API versions | Notes |
+|---------|--------------|-------|
+| X       | v1           |       |
+| X+1     | v1, v2alpha1 |       |
+| X+2     | v1, v2alpha2 | * v2alpha1 is removed, “action required” relnote |
+| X+3     | v1, v2beta1  | * v2alpha2 is removed, “action required” relnote |
+| X+4     | v1, v2beta1, v2beta2 | * v2beta1 is deprecated, “action required” relnote |
+| X+5     | v1, v2, v2beta2      | * v2beta1 is removed, “action required” relnote<br> * v2beta2 is deprecated, “action required” relnote<br> * v1 is deprecated, “action required” relnote |
+| X+6     | v1, v2               | * v2beta2 is removed, “action required” relnote |
+| X+7     | v1, v2               | |
+| X+8     | v1, v2               | |
+| X+9     | v2                   | * v1 is removed, “action required” relnote |
+
+### REST resources (aka API objects)
+
+Consider a hypothetical REST resource named Widget, which was present in API v1
+in the above timeline, and which we want to deprecate.  We [document](http://kubernetes.io/docs/deprecated/)
```

Comment:
release notes seem like a good way to announce them, but I think we need a page that links to current deprecations, no?